ratpack-core/src/test/groovy/ratpack/http/client/HttpClientSmokeSpec.groovy

```diff
@@ -517,4 +519,73 @@ BAR
     text == "bar"
   }
 
+
+  def "can handle compressed responses"() {
+    given:
+    def stringBuf = new ByteArrayOutputStream()
+    def writer = new OutputStreamWriter(new GZIPOutputStream(stringBuf))
+    writer.write("bar")
+    writer.close()
+
+    otherApp {
+      get("foo") {
+        response.headers.add(CONTENT_ENCODING, "gzip")
+        response.send(stringBuf.toByteArray())
+      }
+    }
+
+    and:
+    handlers {
+      get { HttpClient httpClient ->
+        httpClient.request(otherAppUrl("foo")) { RequestSpec rs ->
+          rs.headers.set("Accept-Encoding", "compress, gzip")
```

Comment:
If decompress is on, I wonder whether we should automatically set these headers if they're not already set